### PR TITLE
Jiahui/profile firebase

### PR DESCRIPTION
**Pull request by jchen23-stf**
# *Name of the PR*
Connecting profile page to firebase data 

## :recycle: Current situation & Problem
*Describe the current situation (if possible, with an exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*
- The profile page is not linked with backend data 

## :bulb: Proposed solution
*Describe the solution and how this affects the project and internal structure*

## :gear: Release Notes 
*Add a summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*
- Now the user name, email, and condition are pulled from the firebase database 
- When the user edit their condition, the firebase record is also updated 
- When user signs out, the firebase also signs out 

## :heavy_plus_sign: Additional Information
*Provide some additional information if possible*

### Related PRs
*Reference the related PRs*

### Testing
*Are there tests included? If yes, which situations are tested, and which corner cases are missing?*

### Reviewer Nudging
*Where should the reviewer start? Where is a good entry point?*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jchen23-stf** [comment]
Running into this bug where current_user cannot be accessed after the user logs out, fixing it: https://stackoverflow.com/questions/74406692/firestore-document-path-cannot-be-empty

**Comment by jenlmoore** [comment]
Can you please add some screenshots? Thanks!

File: Utah/Utah.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ import UtahSharedContext
 @main
 struct Utah: App {
     @UIApplicationDelegateAdaptor(UtahAppDelegate.self) var appDelegate
-    @AppStorage(StorageKeys.onboardingFlowComplete) var completedOnboardingFlow = true
+    @AppStorage(StorageKeys.onboardingFlowComplete) var completedOnboardingFlow = false
```
Comment (jenlmoore) [inline Utah/Utah.swift]: Why did you change this to false?

File: UtahModules/Package.swift [hunk 2]
```diff
@@ -47,7 +47,13 @@ let package = Package(
         .target(
             name: "UtahProfile",
             dependencies: [
-                .target(name: "UtahSharedContext")
+                .target(name: "UtahSharedContext"),
+                .product(name: "FHIR", package: "CardinalKit"),
+                .product(name: "Questionnaires", package: "CardinalKit"),
+                .product(name: "Scheduler", package: "CardinalKit"),
```
Comment (jenlmoore) [inline UtahModules/Package.swift]: I dont think you use these in the profile page, if you don't can you please take them out?

**Comment by jchen23-stf** [comment]
Sometimes when the user first sign up the data is not immediately loaded (needed to trigger change for data to load), investigating this issue rn

**Review by BKrajancich — APPROVED** [review]
